### ISSUE TITLE
Package Versioning should be better handled as per semver BNF i.e. Anaconda.Miniconda3

### ISSUE THREAD

**Issue by sarvex** [microsoft/winget-cli#3164]
### Brief description of your issue

Winget does not find the latest version of the software as per [semver](https://semver.org/). Ideally 3.10 should be **greater than** 3.9, but in this case the [Backus Naur Form](https://semver.org/#backusnaur-form-grammar-for-valid-semver-versions) is not implemented properly and thus 3.10 becomes **less than** 3.9

### Steps to reproduce

```powershell
winget show Anaconda.Miniconda3
```

### Expected behavior

```powershell
winget show Anaconda.Miniconda3

Found Miniconda3 [Anaconda.Miniconda3]
Version: py310_23.1.0-1
Publisher: Anaconda, Inc.
Publisher Url: https://www.anaconda.com/
Author: Anaconda
Moniker: miniconda3
Description: Miniconda is a free minimal installer for conda. It is a small, bootstrap version of Anaconda that includes only conda, Python, the packages they depend on, and a small number of other useful packages, including pip, zlib and a few others.
Homepage: https://docs.conda.io/en/latest/miniconda.html
License: BSD 3-Clause License
License Url: https://docs.conda.io/en/latest/license.html
Privacy Url: https://www.anaconda.com/privacy-policy
Copyright: Miniconda - © 2015-2021, Anaconda, Inc. /  Conda - (C) 2017 Continuum Analytics, Inc. (dba Anaconda, Inc.). https://www.anaconda.com. All Rights Reserved
Copyright Url: https://www.anaconda.com/end-user-license-agreement-miniconda
Tags:
  miniconda3
  miniconda
  python
  conda
Installer:
  Installer Type: nullsoft
  Installer Url: https://repo.anaconda.com/miniconda/Miniconda3-py310_23.1.0-1-Windows-x86_64.exe
  Installer SHA256: b33797064593ab2229a0135dc69001bea05cb56a20c2f243b1231213642e260a
```

### Actual behavior

```powershell
winget show Anaconda.Miniconda3

Found Miniconda3 [Anaconda.Miniconda3]
Version: py39_4.10.3
Publisher: Anaconda, Inc.
Publisher Url: https://www.anaconda.com/
Author: Anaconda
Moniker: miniconda3
Description: Miniconda is a free minimal installer for conda. It is a small, bootstrap version of Anaconda that includes only conda, Python, the packages they depend on, and a small number of other useful packages, including pip, zlib and a few others.
Homepage: https://docs.conda.io/en/latest/miniconda.html
License: BSD 3-Clause License
License Url: https://docs.conda.io/en/latest/license.html
Privacy Url: https://www.anaconda.com/privacy-policy
Copyright: Miniconda - © 2015-2021, Anaconda, Inc. /  Conda - (C) 2017 Continuum Analytics, Inc. (dba Anaconda, Inc.). https://www.anaconda.com. All Rights Reserved
Copyright Url: https://www.anaconda.com/end-user-license-agreement-miniconda
Tags:
  miniconda3
  miniconda
  python
  conda
Installer:
  Installer Type: nullsoft
  Installer Url: https://repo.anaconda.com/miniconda/Miniconda3-py39_4.10.3-Windows-x86_64.exe
  Installer SHA256: b33797064593ab2229a0135dc69001bea05cb56a20c2f243b1231213642e260a
```

### Environment

```shell
Windows Package Manager v1.4.10173
Copyright (c) Microsoft Corporation. All rights reserved.

Windows: Windows.Desktop v10.0.22621.1555
System Architecture: X64
Package: Microsoft.DesktopAppInstaller v1.19.10173.0

Logs: %LOCALAPPDATA%\Packages\Microsoft.DesktopAppInstaller_8wekyb3d8bbwe\LocalState\DiagOutputDir

User Settings: %LOCALAPPDATA%\Packages\Microsoft.DesktopAppInstaller_8wekyb3d8bbwe\LocalState\settings.json

Links
---------------------------------------------------------------------------
Privacy Statement   https://aka.ms/winget-privacy
License Agreement   https://aka.ms/winget-license
Third Party Notices https://aka.ms/winget-3rdPartyNotice
Homepage            https://aka.ms/winget
Windows Store Terms https://www.microsoft.com/en-us/storedocs/terms-of-sale
```

**Comment by denelon** [comment]
Duplicate of #2394

* #2394

**Comment by sarvex** [comment]
@danielorn it is not duplicate of #2394. Even if you fix #2394 the problem will persist, as #2394 never mentions .10 > .9 these are two completely different issue.

**Comment by denelon** [comment]
The versions for Miniconda3 are not semver compliant.

If we ignored the "preamble" as the leading portion of the version up to the underscore "_", then we would have semver compliant versions.

Given the output below for the package, it's hard for WinGet to reason about the versions. If the preambles were stripped, we would have:
* 23.1.0-1
* 4.10.3
* 4.9.2
* 4.8.3

```text
PS C:\Users\denelon> winget show Anaconda.Miniconda3 --versions
Found Miniconda3 [Anaconda.Miniconda3]
Version
--------------
py39_4.10.3
py39_4.9.2
py37_4.8.3
py310_23.1.0-1
```

**Comment by sarvex** [comment]
@denelon apologies, you are right, there has been an oversight on my part. 

maybe its better to raise a bug on Anaconda to have consistent naming convention...

**Comment by denelon** [comment]
@sarvex
No worries. 😊
Terminology can be very confusing. WinGet still has plenty of improvements to strive towards better behavior for "non-standard" based "packages". The approach has been to try and meet publishers where they are so we can have a robust ecosystem of software available with deterministic behaviors.

I think making the ask on Anaconda would be the best long-term solution. I do expect they would need time to reason about their current versioning strategy and how best to adopt a semver strategy. WinGet is a relative newcomer to the world of software installation on Windows. It will take time for ISVs (Independent Software Vendors) to fully embrace WinGet and provide their customers with the best experiences.

**Comment by denelon** [comment]
I will also openly point out that `winget --version` also exhibits a non semver behavior where there is a "v" preamble. As others may have already implemented code expecting that inappropriate character, I would treat it as a breaking change that would not likely be fixed until WinGet is ready to move to a 2.0 version.
